packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -64,14 +88,26 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
 
       if (normalTexture != null && normalMap != null) {
         normalTextures.add(normalMap);
+      } else {
+        normalTexture = new EmptyNormalTexture();
+        material.normalMap = new ThreeTexture();
```

Comment:
I don't think you want to do this. We want to create a new texture wrapper, not an actual empty Three.js texture. That could have all sorts of strange side effects, especially for performance (and potentially running out of samplers, if we take this too far). We should add the logic to `setURI()` that will create a texture if one is not present on the three.js material yet. That probably means you'll need to store a reference to the material and not just the correlated textures.

Reply:
Update the PR as per our conversation, externally the API is nice to work with. Internally I'm not so happy with the weird contortions and indirections, it works though, and it does highlight (painfully so) some areas to address in later scene-graph updates.